python/lsst/meas/algorithms/psfDeterminer.py

```diff
@@ -63,6 +72,31 @@ class BasePsfDeterminerTask(pipeBase.Task, metaclass=abc.ABCMeta):
     def __init__(self, config, schema=None, **kwds):
         pipeBase.Task.__init__(self, config=config, **kwds)
 
+    def downsampleCandidates(self, inputCandidateList):
+        """Down-sample candidates from the input candidate list.
+
+        Parameters
+        ----------
+        inputCandidateList : `list` [`lsst.meas.algorithms.PsfCandidate`]
+            Input candidate list.
+
+        Returns
+        -------
+        outputCandidateList : `list` [`lsst.meas.algorithms.PsfCandidate`]
+            Down-selected candidate list.
+        """
+        if len(inputCandidateList) <= self.config.maxCandidates:
+            return inputCandidateList
+
```

Comment:
You could also add the number of PSF candidates used (in either case) to the task metadata. This would allow you to test that it is used correctly in the psf and piff unit tests.

Reply:
I hadn't realized this was already there ... 🤦